system/HTTP/ContentSecurityPolicy.php

```diff
@@ -188,52 +185,47 @@ class ContentSecurityPolicy
 	 */
 	protected $reportOnlyHeaders = [];
 
-	//--------------------------------------------------------------------
-
 	/**
-	 * ContentSecurityPolicy constructor.
+	 * Constructor.
 	 *
 	 * Stores our default values from the Config file.
 	 *
 	 * @param ContentSecurityPolicyConfig $config
 	 */
 	public function __construct(ContentSecurityPolicyConfig $config)
 	{
-		foreach ($config as $setting => $value) // @phpstan-ignore-line
+		foreach (get_object_vars($config) as $setting => $value)
 		{
-			if (isset($this->{$setting}))
+			if (property_exists($this, $setting))
```

Comment:
property_exists with isset is different, property_exists got data even value is null. ref https://3v4l.org/m99JJ

Reply:
Yes. But isn't it a bug to use `isset`? For example, this class' `$reportURI` property is uninitialized so `isset` to it will evaluate to `false` and I cannot set it in the constructor.

Reply:
probably, @MGatner any opinion?

Reply:
Yes let's use `property_exists` so we can override the null values.